### PR TITLE
Dynamic Obstacles Environment

### PR DESCRIPTION
**Pull request by Suman7495**
Based on issue #50

### CODE REVIEW

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<component name="InspectionProjectProfileManager">
```
Comment (maximecb) [inline .idea/inspectionProfiles/Project_Default.xml]: This file shouldn't be included in the PR

File: README.md [hunk 2]
```diff
@@ -19,23 +19,23 @@ Please use this bibtex if you want to cite this repository in your publications:
 
 ```
 @misc{gym_minigrid,
-  author = {Chevalier-Boisvert, Maxime and Willems, Lucas},
+  author = {Chevalier-Boisvert, Maxime and Willems, Lucas and Pal, Suman},
   title = {Minimalistic Gridworld Environment for OpenAI Gym},
   year = {2018},
   publisher = {GitHub},
   journal = {GitHub repository},
-  howpublished = {\url{https://github.com/maximecb/gym-minigrid}},
```
Comment (maximecb) [inline README.md]: Waiiiit a minute ;)

**Comment by maximecb** [comment]
Hi Suman.

The code for your new environment itself seems fine, but there are a few changes that I would like to see:
- You included files in the PR that shouldn't be included, ie: settings files for your IDE
- Changes to the README should be minimal. You changed things such as the URL for the repo in your fork, but we don't want to change that on this repo.
- Your new obstacle has no internal logic, so I would propose simply reusing the `Ball` object instead, and preventing the pickup action in your environment (if action == self.actions.pickup: return ...). That way we can avoid adding a new type of object with a new observation encoding, etc.

**Comment by Suman7495** [comment]
While implementing the 3rd change, I noticed something strange: no matter which environment, the ``action_space`` has 7 available actions, including ``pickup`` etc. So for the ``Empty`` environment, during the RL training, the agent can select from 7 actions, while actually to achieve the goal it requires only 3 actions. 

This may not cause a difference for sparse rewards, but otherwise, it increases the complexity unnecessarily. Instead would it not be more optimal to have a method in each environment which constrains the ``action_space`` to only the required actions?

**Comment by maximecb** [comment]
Hi Suman,

You are right. The action space has actions that aren't needed, which is somewhat inefficient. I wanted to keep the actions mapped to the same values so we could do experiments with transfer learning or curriculum learning, that is, train in one environment, then test the model in a different environment. For that, the action spaces need to be compatible.

That being said, if you only need the left/right/back actions and not pickup/drop/etc, you should be able to disable the other actions in your environment this way, in the constructor:

```self.action_space = spaces.Discrete(self.actions.forward+1)```

Other remarks about this pull request:

- You've changed the `reward_range` in `minigrid.py`. This change should be made in your environment rather than in the base class. Ideally, no changes should be made in `minigrid.py`

- Revert/remove changes to `run_tests.py` and `manual_control.py`.  Take a look at the "Files Changed" tab the pull request to see all the changes you've made.

**Comment by maximecb** [comment]
Ping @Suman7495

**Comment by Suman7495** [comment]
Hi Maxime,

Sorry for the delay. Unfortunately this week I'm a little caught up with work. I'll try to make the required changes on the weekend and contact you after that.

**Comment by Suman7495** [comment]
Hi Maxime,

I have made the required changes. Kindly let me know if any other details are to be mended.

Thank you.

**Comment by maximecb** [comment]
It looks good. If you can revert all changes to `gym_minigrid/minigrid.py` and `run_tests.py` I think it's ready to merge. Look under files changed to see the changes that were made to those files.

**Comment by Suman7495** [comment]
Hi Maxime,

I did a final check, and now I think all the changes are reverted.

**Comment by maximecb** [comment]
Hi, you still have a number of one-line changes in minigrid.py and run_tests.py. Can you please copy the versions of these files from the master branch of this repository and commit that?

**Comment by Suman7495** [comment]
I downloaded the original files from the master and committed them to my branch. I think it should be fine now.

**Comment by maximecb** [comment]
Thank you.
